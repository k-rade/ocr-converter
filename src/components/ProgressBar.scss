.progress-bar {
  height: 4px;
  width: 100%;
  background-color: var(--color-gray-700);
  transition: all 0.15s ease;
  position: relative;
  display: flex;
  margin-top: 0.5rem;

  &-line {
    width: 100%;
    height: 100%;
    background-color: var(--color-primary-600);
    position: absolute;
    transition: width 0.25s;
  }

  &-container {
    margin: 0.5rem;
    width: 100%;
    height: 8px;
    background-color: transparent;

    &-progress {
      font-weight: 600;
      font-size: 0.875rem;
    }
  }

  &-txt {
    margin: 0.75rem 0;
    font-size: 0.875rem;
  }
}