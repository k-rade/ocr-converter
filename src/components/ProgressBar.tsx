import "./ProgressBar.scss";

interface Props {
  progress: number;
}

const ProgressBar = ({ progress }: Props): JSX.Element => (
  <div className="progress-bar-container">
    <span className="progress-bar-container-progress">
      {Math.round(progress * 100)}%
    </span>
    <div className="progress-bar">
      <div className="progress-bar-line"></div>
    </div>
    <p className="progress-bar-txt">Converting in progress</p>
  </div>
);

export default ProgressBar;
