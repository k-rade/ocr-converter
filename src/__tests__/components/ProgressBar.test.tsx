import { render } from "@testing-library/react";
import ProgressBar from "../../components/ProgressBar";

describe("ProgressBar", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<ProgressBar progress={0.52} />);

    expect(baseElement).toMatchSnapshot();
  });
});
