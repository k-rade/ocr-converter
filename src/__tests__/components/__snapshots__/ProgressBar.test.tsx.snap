// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render 1`] = `
<body>
  <div>
    <div
      class="progress-bar-container"
    >
      <span
        class="progress-bar-container-progress"
      >
        52
        %
      </span>
      <div
        class="progress-bar"
      >
        <div
          class="progress-bar-line"
          style="width: 52%;"
        />
      </div>
      <p
        class="progress-bar-txt"
      >
        Converting in progress...
      </p>
    </div>
  </div>
</body>
`;
